LEGO1/legoinputmanager.cpp

```diff
@@ -46,19 +46,27 @@ LegoInputManager::~LegoInputManager()
 	Destroy();
 }
 
+// OFFSET: LEGO1 0x1005b960
+void LegoInputManager::Create()
+{
+	// TODO
+	if (m_eventQueue == NULL)
+		m_eventQueue = new LegoEventQueue();
+}
+
 // OFFSET: LEGO1 0x1005bfe0
 void LegoInputManager::Destroy()
 {
 	ReleaseDX();
 
+	if (m_unk0x5c)
+		delete m_unk0x5c;
+	m_unk0x5c = NULL;
```

Comment:
Did the developers actually do this? I don't think its necessary to set something to NULL after deletion

Reply:
Yeah, it's in there:
```assembly
    1005bfe8 8b 4e 5c   MOV     ECX,dword ptr [ESI + 0x5c]
    1005bfeb 85 c9      TEST    ECX,ECX
    1005bfed 74 06      JZ      LAB_1005bff5
    1005bfef 6a 01      PUSH    0x1
    1005bff1 8b 01      MOV     EAX,dword ptr [param_1]
    1005bff3 ff 10      CALL    dword ptr [EAX]    ; `scalar deleting destructor'
                    LAB_1005bff5                      XREF[1]: 1005bfed(j)  
```

`delete` or `free()` doesn't set the pointer to NULL, so you'll segfault if you dereference it again. Setting to NULL allows the code to ask "is there something here?" and if the pointer is NULL then the answer is "no."